client/state/data-layer/wpcom/sites/comments/change-status/index.js

```diff
@@ -27,24 +29,30 @@ const changeCommentStatus = ( { dispatch }, action ) => {
 					status,
 				},
 			},
-			action
+			{
+				...action,
+				previousStatus: get( comment, 'status' ),
+			}
 		)
 	);
 };
 
 const changeCommentStatusSuccess = ( { dispatch }, action, next, data ) => {
-	dispatch( {
-		...action,
-		status: get( data, 'status' ),
-	} );
+	dispatch(
+		local( {
+			...action,
+			status: get( data, 'status' ),
+		} )
+	);
```

Comment:
we may want to dispatch this _only if_ the data in the response from the API call doesn't match what we find in state…

```js
const fromState = getCommentStatus( getState(), action.siteId, action.commentId );
const fromServer = data.comment_status;

if ( fromState !== fromServer && data.status ) {
	dispatch( local( {
		...action,
		status: data.status,
	} ) );
}
```